pkg/collector/corechecks/snmp/config.go

```diff
@@ -0,0 +1,271 @@
+	Profiles      profileConfigMap `yaml:"profiles"`
+	GlobalMetrics []metricsConfig  `yaml:"global_metrics"`
+}
+
+type snmpInstanceConfig struct {
+	IPAddress        string            `yaml:"ip_address"`
+	Port             Number            `yaml:"port"`
+	CommunityString  string            `yaml:"community_string"`
+	SnmpVersion      string            `yaml:"snmp_version"`
+	Timeout          Number            `yaml:"timeout"`
+	Retries          Number            `yaml:"retries"`
+	User             string            `yaml:"user"`
+	AuthProtocol     string            `yaml:"authProtocol"`
+	AuthKey          string            `yaml:"authKey"`
+	PrivProtocol     string            `yaml:"privProtocol"`
+	PrivKey          string            `yaml:"privKey"`
+	ContextName      string            `yaml:"context_name"`
+	Metrics          []metricsConfig   `yaml:"metrics"`
+	MetricTags       []metricTagConfig `yaml:"metric_tags"`
+	Profile          string            `yaml:"profile"`
+	UseGlobalMetrics bool              `yaml:"use_global_metrics"`
+}
+
+type snmpConfig struct {
+	ipAddress         string
+	port              uint16
+	communityString   string
+	snmpVersion       string
+	timeout           int
+	retries           int
+	user              string
+	authProtocol      string
+	authKey           string
+	privProtocol      string
+	privKey           string
+	contextName       string
+	oidConfig         oidConfig
+	metrics           []metricsConfig
+	metricTags        []metricTagConfig
+	oidBatchSize      int
+	profiles          profileDefinitionMap
+	profileTags       []string
+	uptimeMetricAdded bool
+}
+
+func (c *snmpConfig) refreshWithProfile(profile string) error {
+	if _, ok := c.profiles[profile]; !ok {
+		return fmt.Errorf("unknown profile `%s`", profile)
+	}
+	log.Debugf("Refreshing with profile `%s` with content: %#v", profile, c.profiles[profile])
+	tags := []string{"snmp_profile:" + profile}
+	definition := c.profiles[profile]
+
+	c.metrics = append(c.metrics, definition.Metrics...)
+	c.metricTags = append(c.metricTags, definition.MetricTags...)
+	c.oidConfig.scalarOids = append(c.oidConfig.scalarOids, parseScalarOids(definition.Metrics, definition.MetricTags)...)
+	c.oidConfig.columnOids = append(c.oidConfig.columnOids, parseColumnOids(definition.Metrics)...)
+
+	if definition.Device.Vendor != "" {
+		tags = append(tags, "device_vendor:"+definition.Device.Vendor)
+	}
+	c.profileTags = tags
+	return nil
+}
+
+func (c *snmpConfig) addUptimeMetric() {
+	if c.uptimeMetricAdded {
+		return
+	}
+	metricConfig := getUptimeMetricConfig()
+	c.metrics = append(c.metrics, metricConfig)
+	c.oidConfig.scalarOids = append(c.oidConfig.scalarOids, metricConfig.Symbol.OID)
+	c.uptimeMetricAdded = true
+}
+
+func (c *snmpConfig) getStaticTags() []string {
+	tags := []string{"snmp_device:" + c.ipAddress}
+	return tags
+}
+
+// toString used for logging, it will hide sensitive information
+func (c *snmpConfig) toString() string {
+	configCopy := *c
```

Comment:
Agree it might not be safe to use shallow copy here. 

I made the list of fields to stringily here: https://github.com/DataDog/datadog-agent/pull/6749/commits/c582e592c970ca35d946a918f04896e97b55d655